code/game/machinery/newscaster/newspaper.dm

```diff
@@ -10,167 +15,172 @@
+			if(times_censored[i] > 0)
 				return TRUE
 			else
 				return FALSE
 	return FALSE
 
-/obj/item/newspaper/Topic(href, href_list)
-	var/mob/living/U = usr
-	..()
-	if((src in U.contents) || (isturf(loc) && in_range(src, U)))
-		U.set_machine(src)
-		if(href_list["next_page"])
-			if(curr_page == pages+1)
-				return //Don't need that at all, but anyway.
-			if(curr_page == pages) //We're at the middle, get to the end
-				screen = 2
-			else
-				if(curr_page == 0) //We're at the start, get to the middle
-					screen=1
-			curr_page++
-			playsound(loc, SFX_PAGE_TURN, 50, TRUE)
-		else if(href_list["prev_page"])
-			if(curr_page == 0)
-				return
-			if(curr_page == 1)
-				screen = 0
-			else
-				if(curr_page == pages+1) //we're at the end, let's go back to the middle.
-					screen = 1
-			curr_page--
-			playsound(loc, SFX_PAGE_TURN, 50, TRUE)
-		if(ismob(loc))
-			attack_self(loc)
+/obj/item/newspaper/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Newspaper", name)
+		ui.open()
 
-/obj/item/newspaper/attackby(obj/item/W, mob/living/user, params)
-	if(burn_paper_product_attackby_check(W, user))
+/obj/item/newspaper/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
 		return
 
-	if(istype(W, /obj/item/pen))
-		if(!user.can_write(W))
-			return
-		if(scribble_page == curr_page)
-			to_chat(user, span_warning("There's already a scribble in this page... You wouldn't want to make things too cluttered, would you?"))
+	switch(action)
+		if("next_page")
+			//We're at the very end, nowhere else to go.
+			if(current_page == news_content.len + 1)
+				return TRUE
+			current_page++
+		if("prev_page")
+			//We haven't started yet, nowhere else to go.
+			if(!current_page)
+				return TRUE
+			current_page--
 		else
-			var/s = tgui_input_text(user, "Write something", "Newspaper")
-			if (!s)
-				return
-			if(!user.can_perform_action(src))
-				return
-			scribble_page = curr_page
-			scribble = s
-			attack_self(user)
-			add_fingerprint(user)
-	else
-		return ..()
+			return TRUE
+	SStgui.update_uis(src)
+	playsound(loc, SFX_PAGE_TURN, 50, TRUE)
+	return TRUE
+
+/obj/item/newspaper/ui_static_data(mob/user)
+	var/list/data = list()
+	data["channels"] = list()
+	for(var/datum/feed_channel/news_channels as anything in news_content)
+		data["channels"] += list(list(
+			"name" = news_channels.channel_name,
+			"page_number" = news_content.Find(news_channels),
+		))
+	return data
+
+/obj/item/newspaper/ui_data(mob/user)
+	var/list/data = list()
+	data["current_page"] = current_page
+	data["scribble_message"] = (scribble_page == current_page) ? scribble_text : null
+	if(saved_wanted_icon)
+		user << browse_rsc(saved_wanted_icon, "wanted_photo.png")
+	data["wanted_information"] = list(list(
+		"wanted_criminal" = saved_wanted_criminal,
+		"wanted_body" = saved_wanted_body,
+		"wanted_photo" = (saved_wanted_icon ? "wanted_photo.png" : null),
+	))
```

Comment:
Looking at it, it actually never seems to be more than one element long, is that intentional? Why even have it be a list, futureproofing?